### ISSUE TITLE
Need clarification on ASSERT.TXT file appearing on K64F rev 0241

### ISSUE THREAD

**Issue by VeliMattiLahtela** [mbedmicro/DAPLink#102]
Is this something major that we should be concerned about or is it something trivial that does not affect behaviour of the board.

No documentation on this issue.

The contents of the file:

```
cat /media/pi/DAPLINK1/ASSERT.TXT
Assert
File: ..\..\..\source\usb\msc\usbd_msc.c
Line: 237
Source: Application

```

Source application, does that tell that the application has hard faulted or something?

**Comment by c1728p9** [comment]
Hi @VeliMattiLahtela, this assert indicates that more data than expected was buffered on a write event.  In this case this was a recoverable error, so the device didn't crash or hard fault.  If you can provide steps to reproduce this problem it would greatly help me in fixing this.

Here is the line this error corresponds to:
https://github.com/mbedmicro/DAPLink/blob/34182e2cce4ca99073443ef29fbcfaab9e18caec/source/usb/msc/usbd_msc.c#L237

**Comment by c1728p9** [comment]
I was able to reproduce this locally and found the root cause to be a logic error in the assert. This should be fixed by #219.
